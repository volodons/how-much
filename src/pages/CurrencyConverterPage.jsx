import { CurrencyConverter, Navigation, Title } from '../components/index';

const CurrencyConverterPage = () => {
    return (
        <>
            <Navigation />
            <Title />
            <CurrencyConverter />
        </>
    );
};

export default CurrencyConverterPage;
