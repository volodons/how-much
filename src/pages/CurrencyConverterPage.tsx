import React from 'react';

import { CurrencyConverter, Navigation, Title } from '../components';

const CurrencyConverterPage: React.FC = () => (
    <>
        <Navigation />
        <Title />
        <CurrencyConverter />
    </>
);

export default CurrencyConverterPage;
