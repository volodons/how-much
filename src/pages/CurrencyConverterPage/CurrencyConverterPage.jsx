import CurrencyConverter from "../../components/CurrencyConverter/CurrencyConverter";

const CurrencyConverterPage = () => {
    return <CurrencyConverter />;
};

export default CurrencyConverterPage;
