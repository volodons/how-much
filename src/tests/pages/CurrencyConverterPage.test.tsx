import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../../redux';
import CurrencyConverterPage from '../../pages/CurrencyConverterPage';

describe('CurrencyConverterPage component', () => {
    it('should be a functional component', () => {
        expect(CurrencyConverterPage).toBeInstanceOf(Function);
    });

    it('should render Navigation component', () => {
        render(
            <Provider store={store}>
                <CurrencyConverterPage />
            </Provider>
        );
        const navigationComponent = screen.getByTestId('navigation-component');
        expect(navigationComponent).toBeInTheDocument();
    });

    it('should render Title component', () => {
        render(
            <Provider store={store}>
                <CurrencyConverterPage />
            </Provider>
        );
        const titleComponent = screen.getByTestId('title-component');
        expect(titleComponent).toBeInTheDocument();
    });

    it('should render CurrencyConverter component', () => {
        render(
            <Provider store={store}>
                <CurrencyConverterPage />
            </Provider>
        );
        const currencyConverterComponent = screen.getByTestId(
            'currency-converter-component'
        );
        expect(currencyConverterComponent).toBeInTheDocument();
    });

    it('should match a snapshot', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <CurrencyConverterPage />
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
