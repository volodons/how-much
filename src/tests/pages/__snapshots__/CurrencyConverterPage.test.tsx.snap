// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyConverterPage component should match a snapshot 1`] = `
<div
  className="MuiBox-root css-0"
  data-testid="currency-converter-page"
>
  <div
    className="MuiBox-root css-0"
    data-testid="navigation-component"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1sg6drv-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-188uhum-MuiSvgIcon-root"
        data-testid="open-button"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
        />
      </svg>
    </button>
  </div>
  <h1
    className="MuiTypography-root MuiTypography-h1 css-1dha1by-MuiTypography-root"
    data-testid="title-component"
  >
    How Much?
  </h1>
  <div
    className="MuiBox-root css-0"
    data-testid="currency-converter-component"
  >
    <div
      className="MuiBox-root css-15z49vq"
    >
      <span
        className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-16yyvdj-MuiCircularProgress-root"
        role="progressbar"
        style={
          Object {
            "height": "5rem",
            "width": "5rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </span>
    </div>
  </div>
</div>
`;
